### ISSUE TITLE
Combined schema support - oneOf key

### ISSUE THREAD

**Issue by Karandashyk** [apiaryio/dredd#1680]
**Describe the problem**
I am facing the issue with **oneOf** key, which is an important feature in the specification I'm testing. 
Both in **requestBody** and **responses** schemas are defined with a oneOf keyword. I was able to write a converter that allows to test cases with allOf and referenced schemas (with $ref). But oneOf behavior cannot be rewritten differently with the help of supported features, and I have no idea how to go around its functionality.

**Expected behavior**
I would like Dredd to run transactions for all **oneOf** schemas (both make transactions with different requestBody data and for different expected responses), but it is not clear how Dredd will support the key, currently one transaction is generated for a distinct path.  

Are there any thoughts how oneOf support can be implemented? Any ideas, even some hook-tricks will be helpful (was already thinking about modifying Transactions Object before the test - in beforeAll hook transactions can be complimented with additional custom ones). 
Will it be possible to run couple of transactions in for oneOf in native Dredd? Are there any plans regarding feature support? 

As for example, classic **oneOf** case in request:
``` 
paths:
  /pets:
    patch:
      requestBody:
        content:
          application/json:
            schema:
              oneOf:
                - $ref: '#/components/schemas/Cat'
                - $ref: '#/components/schemas/Dog'
``` 
Respectively, I would like to run two transactions: for Cat and for Dog schema.

**Comment by kylef** [comment]
> Are there any plans regarding feature support?

Support for oneOf use in generation of request bodies has been introduced in https://github.com/apiaryio/api-elements.js/pull/500 which will trickle into a future Dredd update.

**Comment by kylef** [comment]
This is now released in Dredd 13.1.2.
